sphinx_gallery/scrapers.py

```diff
@@ -110,6 +113,17 @@ def matplotlib_scraper(block, block_vars, gallery_conf, **kwargs):
     from matplotlib.animation import Animation
     image_path_iterator = block_vars['image_path_iterator']
     image_rsts = []
+    # Check for srcset hidpi images
+    srcset = gallery_conf.get('image_srcset', [""])
```

Comment:
```suggestion
    srcset = gallery_conf.get('image_srcset')
    if not srcset:
        srcset = [""]
```
This also catches the case of `gallery_conf['image_srcset'] = []`. AFAICT an emtpy  `srcset` is not valid in HTML, so we should handle this. The alternative would be to remove the `srcset` attribute from `<img>`, but that's more work and I don't think it can do harm if it's present with the 1x data.

Reply:
Just to be clear: `""` translates simply to  `:srcset: images/myimage.png`

Reply:
I think that's ok https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attr-srcset.

Reply:
I guess we could remove the explicit "" and just assume it is implied.  And sure we could not include the srcset tag if it's empty.

Reply:
Whichever works for you,  as long as we don't insert an empty parameter in HTML.